lib/validators/SolanaTxValidator.js

```diff
@@ -0,0 +1,221 @@
+      console.log(errorMessage)
+      return { valid: false, markAs: "cancelled", error: errorMessage, switchHWToManualMode: true }
+    }
+
+    return { valid: true }
+  }
+
+  isSplTxValid(blockchainTransaction, txn) {
+    const amountInBaseUnit = new BigNumber(blockchainTransaction.amount)
+
+    if (!amountInBaseUnit.isEqualTo(txn.amount)) {
+      const errorMessage = `Incorrect arguments. Total amount is not equal to transaction amount`
+      console.log(errorMessage)
+      return { valid: false, markAs: "cancelled", error: errorMessage, switchHWToManualMode: true }
+    }
+
+    return { valid: true }
+  }
+
+  isSplTxBatchValid(blockchainTransaction, txn) {
+    const amountInBaseUnit = new BigNumber(blockchainTransaction.amount)
+
+    // validate that tx tables size equals to tx count in order to be able to cancel dublicate transaction
+    if ((!Array.isArray(blockchainTransaction.blockchainTransactables) ||
+      !Array.isArray(txn.amount) || !Array.isArray(txn.to) ||
+      blockchainTransaction.blockchainTransactables.length != txn.amount.length &&
+      blockchainTransaction.blockchainTransactables.length != txn.to.length)) {
+      const errorMessage = `Incorrect arguments. Please recreate batch and try again`
+      console.log(errorMessage)
+      return { valid: false, markAs: "cancelled", error: errorMessage, switchHWToManualMode: true }
+    }
+
+    // validate that total amount == sum of tx amounts
+    if (!amountInBaseUnit.isEqualTo(BigNumber.sum.apply(null, txn.amount))) {
+      const errorMessage = `Incorrect arguments. Total amount is not equal to transactions sum`
+      console.log(errorMessage)
+      return { valid: false, markAs: "cancelled", error: errorMessage, switchHWToManualMode: true }
+    }
+
+    return { valid: true }
+  }
+
+  isSplFundReleaseScheduleTxValid(blockchainTransaction, txn) {
+    let result = this.isSplTxValid(blockchainTransaction, txn)
+    if (!result.valid) {
+      return result
+    }
+
+    if (isNaN(txn.scheduleId) || parseInt(txn.scheduleId) < 0) {
+      const errorMessage = `Unsupported fund release scheduleId parameter: ${txn.scheduleId}`
+      console.log(errorMessage)
+      return { valid: false, markAs: "cancelled", error: errorMessage, switchHWToManualMode: true }
+    }
+    if (txn.cancelableBy.length > 10) {
+      const errorMessage = `Cancalable by accounts more than 10: ${txn.cancelableBy.length}`
+      console.log(errorMessage)
+      return { valid: false, markAs: "cancelled", error: errorMessage, switchHWToManualMode: true }
+    }
+
+    return { valid: true }
+  }
+
+  isSplBatchFundReleaseScheduleTxValid(blockchainTransaction, txn) {
+    // validate that tx tables size equals to tx count in order to be able to cancel dublicate transaction
+    // check array size equals for all parameters
+    if (!Array.isArray(blockchainTransaction.blockchainTransactables) ||
```

Comment:
Thanks. It is interesting solution. But I am not sure if we need to add extra library just for checking isArray... Because some parameters can be string|number|bignumber and all of them is valid.